src/Timers.cpp

```diff
@@ -486,10 +1104,10 @@ void Timers::TimerUpdates()
   {
     if (m_timers.at(i).iUpdateState == VU_UPDATE_STATE_NONE)
     {
-      XBMC->Log(LOG_INFO, "%s Removed timer: '%s', ClientIndex: '%d'", __FUNCTION__, m_timers.at(i).strTitle.c_str(), m_timers.at(i).iClientIndex);
-      m_timers.erase(m_timers.begin()+i);
-      i=0;
-      iRemoved++;
+        XBMC->Log(LOG_INFO, "%s Removed timer: '%s', ClientIndex: '%d'", __FUNCTION__, m_timers.at(i).strTitle.c_str(), m_timers.at(i).iClientIndex);
+        m_timers.erase(m_timers.begin()+i);
+        i=0;
+        iRemoved++;
```

Comment:
You should rewrite all (!) those index based for loops to range based for loops. Then, you get rid of those ugly and inperformant  `.at(i)` stuff.

Example: `for (unsigned int i=0; i<m_timers.size(); i++)` => `for (`const auto& timer : m_timers)`